packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -364,168 +389,171 @@ contract OPContractsManagerGameTypeAdder is OPContractsManagerBase {
-            IFaultDisputeGame fdg = IFaultDisputeGame(address(getGameImplementation(dgf, GameTypes.CANNON)));
-            IPermissionedDisputeGame pdg =
-                IPermissionedDisputeGame(address(getGameImplementation(dgf, GameTypes.PERMISSIONED_CANNON)));
 
-            // All chains must have a permissioned game, but not all chains must have a fault dispute game.
-            // Whether a chain has a fault dispute game determines how many AddGameInput objects are needed.
-            bool hasFDG = address(fdg) != address(0);
-
-            OPContractsManager.AddGameInput[] memory inputs = new OPContractsManager.AddGameInput[](hasFDG ? 2 : 1);
-            OPContractsManager.AddGameInput memory pdgInput;
-            OPContractsManager.AddGameInput memory fdgInput;
-
-            // Get the existing game parameters and init bond for the permissioned game
-            IFaultDisputeGame.GameConstructorParams memory pdgParams =
-                getGameConstructorParams(IFaultDisputeGame(address(pdg)));
-            uint256 initBond = dgf.initBonds(GameTypes.PERMISSIONED_CANNON);
-
-            string memory saltMixer = reusableSaltMixer(_prestateUpdateInputs[i]);
-            // Create game input with updated prestate but same other params
-            pdgInput = OPContractsManager.AddGameInput({
-                disputeAbsolutePrestate: _prestateUpdateInputs[i].absolutePrestate,
-                saltMixer: saltMixer,
-                systemConfig: _prestateUpdateInputs[i].systemConfigProxy,
-                proxyAdmin: _prestateUpdateInputs[i].proxyAdmin,
-                delayedWETH: IDelayedWETH(payable(address(pdgParams.weth))),
-                disputeGameType: pdgParams.gameType,
-                disputeMaxGameDepth: pdgParams.maxGameDepth,
-                disputeSplitDepth: pdgParams.splitDepth,
-                disputeClockExtension: pdgParams.clockExtension,
-                disputeMaxClockDuration: pdgParams.maxClockDuration,
-                initialBond: initBond,
-                vm: pdgParams.vm,
-                permissioned: true
-            });
-
-            // If a fault dispute game exists, create a new game with the same parameters but updated prestate.
-            if (hasFDG) {
-                // Get the existing game parameters and init bond for the fault dispute game
-                IFaultDisputeGame.GameConstructorParams memory fdgParams =
-                    getGameConstructorParams(IFaultDisputeGame(address(fdg)));
-                initBond = dgf.initBonds(GameTypes.CANNON);
-
-                // Create game input with updated prestate but same other params
-                fdgInput = OPContractsManager.AddGameInput({
+            // Create an array of all of the potential game types to update.
+            GameType[] memory gameTypes = new GameType[](4);
+            gameTypes[0] = GameTypes.CANNON;
+            gameTypes[1] = GameTypes.PERMISSIONED_CANNON;
```

Comment:
The absolute prestate for super games is different, and incompatible, with the absolute prestate of non-super games, and vice versa. I think it'll be better to have a separate `addGameType` function for super games. Or add a discriminant to the call input so we don't accidentally use the same prestate for both game variants.